File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        #Dictionary to store instances
        count_map = {}

        #Iterate through array
        for num in arr:
            count_map[num] = count_map.get(num, 0) + 1

        seen_counts = set()

        for count in count_map.values():
            if count in seen_counts:
                return False

            seen_counts.add(count)

        return True

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
import java.util.Collection;

class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        //Count occurences
        Map<Integer, Integer> countMap = new HashMap<>();
        for(int num: arr){
            countMap.put(num, countMap.getOrDefault(num, 0) + 1);
        }
        //Obtain values
        Collection<Integer> counts = countMap.values();

        //Set of values
        Set<Integer> uniqueCounts = new HashSet<>(counts);
        //COmpare sizes
        return counts.size() == uniqueCounts.size();
    }
}